src/main/java/athleticli/data/Goal.java

```diff
@@ -52,7 +52,7 @@ public Timespan getTimespan() {
      * @param date  The date to be matched.
      * @return      Whether the date is between the timespan.
      */
-    public boolean checkDate(LocalDate date) {
+    public static boolean checkDate(LocalDate date, Timespan timespan) {
```

Comment:
I made this one static @skylee03 as this was easier to adapt to my existing code. Please let me know if it should not be static. Then I will adjust my code accordingly

Reply:
> I made this one static @skylee03 as this was easier to adapt to my existing code. Please let me know if it should not be static. Then I will adjust my code accordingly

I think it's OK, but the JavaDoc also needs to be modified.